courses/views/v2/__init__.py

```diff
@@ -0,0 +1,30 @@
+"""
+Course API Views version 2
+"""
+from django_filters.rest_framework import DjangoFilterBackend
+from rest_framework import viewsets
+from rest_framework.pagination import PageNumberPagination
+
+from courses.models import Program
+from courses.serializers.v2.programs import ProgramSerializer
+
+
+class Pagination(PageNumberPagination):
+    """Paginator class for infinite loading"""
+
+    page_size = 12
+    page_size_query_param = "page_size"
+    max_page_size = 100
+    ordering = "-created_on"
+
+
+class ProgramViewSet(viewsets.ReadOnlyModelViewSet):
+    """API viewset for Programs"""
+
+    permission_classes = []
+
+    serializer_class = ProgramSerializer
+    pagination_class = Pagination
+    filter_backends = [DjangoFilterBackend]
+    filterset_fields = ["id", "live", "readable_id"]
+    queryset = Program.objects.filter().prefetch_related("departments")
```

Comment:
There's probably some other prefetches we could be doing but it looks like that's probably a taller order because a lot of it is inlined into the model class itself.

Reply:
Yeah - the relationships run fairly deep and I ended up adding more queries than saving. I can look at adding some more to the requirements_tree ... tree, which might help some of this, but overall we're still saving a lot with the removal of the courses model from here.  I added 1 per course by adding flexpricingform in, too, but that was to be expected as it wasn't called prior.